engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/sink/TableSink.scala

```diff
@@ -61,7 +60,9 @@ class TableSink(
 
     val inputValueTable = tableEnv
       .fromDataStream(streamOfRows)
-      .select(tableDefinition.sinkRowDataType.getLogicalType.toRowTypeUnsafe.getFieldNames.asScala.toList.map($): _*)
+      .select(
+        tableDefinition.schema.toSinkRowDataType.getLogicalType.toRowTypeUnsafe.getFieldNames.asScala.toList.map($): _*
```

Comment:
1. Doesn't it have to be sorted like in table source?
2. Thats a pretty long expression and its used in similar forms in other places. Maybe its worth to extract some of it to the `TableDefinition` to increase readability and deduplicate some code?

Reply:
1. No, it doesn't. We should insert columns in the same order as in sink. This sorting is only needed when a Row will be serialized using TestInfo prepared by our detection
2. I extracted fields in `TableDefinition` and added some extensions to make these lines shorter